RecoEcal/EgammaCoreTools/src/EcalClusterLazyTools.cc

```diff
@@ -257,11 +257,11 @@ float EcalClusterLazyToolsBase::eseffsirir(const reco::SuperCluster &cluster)
   if (!(fabs(cluster.eta()) > 1.6 && fabs(cluster.eta()) < 3.)) return 0.;
 
   const CaloSubdetectorGeometry *geometryES = geometry_->getSubdetectorGeometry(DetId::Ecal, EcalPreshower);
-  CaloSubdetectorTopology *topology_p = 0;
-  if (geometryES) topology_p = new EcalPreshowerTopology(geometry_);
+  std::unique_ptr<CaloSubdetectorTopology> topology_p;
+  if (geometryES) topology_p.reset(new EcalPreshowerTopology(geometry_));
```

Comment:
These three lines are repeated three times in this file. topology_p is allocated and de-allocated each time one of these methods is called, which may be a lot of unnecessary calls. `geometry_->getSubdetectorGeometry()` is a `const` method, so it is not changing `geometry_`, and its parameters are constant `enums`. It looks to me like topology_p could be made a data member, and these three lines could be moved to the constructor. Would this change cause any problems?